monitor/dbt_project/macros/get_tests_sample_data.sql

```diff
@@ -40,12 +41,55 @@
             {% do elementary_tests_allowlist_status.append('pass') %}
         {% endif %}
         {%- if (test_type == 'dbt_test' and status in ['fail', 'warn']) or (test_type != 'dbt_test' and status in elementary_tests_allowlist_status) -%}
-            {% set test_rows_sample = elementary_internal.get_test_rows_sample(test_results_query, test_type, metrics_sample_limit) %}
+            {# Dimension anomalies return multiple dimensions for the test rows sample, and needs to be handle differently. #}
+            {# Currently we show only the anomalous for all of the dimensions. #}
+            {% if test_sub_type == 'dimension' %}
+                {% set dimension_test_result_query %}
+                    select *
+                    from ({{test_results_query}})
+                    where is_anomalous
+                {% endset %}
+                {% set test_rows_sample = elementary_internal.get_test_rows_sample(dimension_test_result_query, test_type, metrics_sample_limit) %}
+                {% set anomalous = [] %}
+                {% set headers = [{'id': 'elementary_end_time', 'display_name': 'timestamp', 'type': 'date'}] %}
+                {% for sample in test_rows_sample %}
+                    {% set anomalous_sample = {
+                        'elementary_end_time': sample['end_time'],
```

Comment:
I use elementary prefix to prevent the edge case where the customers have dimensions that matches our columns